app/src/main/java/cloudcity/CloudCityParamsRepository.java

```diff
@@ -151,4 +155,35 @@ public void putStringKey(String key, String value) {
                 .putString(key, value)
                 .commit();
     }
+
+    private String getServerURLBasedOnBuildVariant() {
+        String serverUrl = null;
+        if(BuildConfig.IS_DEMO) {
+            serverUrl = DEMO_SERVER_URL;
+        } else if (BuildConfig.IS_STAGING) {
+            serverUrl = STAGING_SERVER_URL;
+        } else if (BuildConfig.IS_PRODUCTION) {
+            serverUrl = PRODUCTION_SERVER_URL;
+        } else {
+            throw new IllegalStateException("Which server environment are we trying to use? Something has went wrong here.");
+        }
+
+        return serverUrl;
+    }
+
+    private String getServerTokenBasedOnBuildVariant() {
+        // All of these tokens are Shark's tokens
+        String serverToken = null;
+        if(BuildConfig.IS_DEMO) {
+            serverToken = "4|VpmKNeLkoFtZbJFTIRVpVWdclzf7LiL9sp83JuVw91ed224b";
+        } else if (BuildConfig.IS_STAGING) {
+            serverToken = "252|GI2CTemW2EX3TmQaDmCzYg1xrs3VEDslnGrfCwp245a84b22";
+        } else if (BuildConfig.IS_PRODUCTION) {
+            serverToken = "41|lAp3yiiWJH3D3ftR54seY6oMO8EEjpees7Y3oJI63b71a1d3";
+        } else {
+            throw new IllegalStateException("Which server environment are we trying to use? Something has went wrong here.");
+        }
+
+        return serverToken;
+    }
```

Comment:
In case of a PI the token will be deleted on the server-side and the problem will disappear. the suggestion is valid, but not super applicable to the current situation.

Will add a TODO to get rid of them though; for now it's not a huge concern.

Reply:
added. resolving.